importer/Cityvizor.Importer.Convertor/Kxx/KxxParser.cs

```diff
@@ -0,0 +1,530 @@
+            Organization: organization,
+            ShouldGive: shouldGive.Value,
+            Gave: gave.Value);
+    }
+
+    /// <summary>
+    ///  Represents G/$ line in .kxx file - document balance description
+    /// G/$rrrrccccccccctttttttttttttttttttttttttttttttttttttttt...
+    /// </summary>
+    /// <param name="input"></param>
+    /// <returns></returns>
+    internal KxxDocumentBalanceDescription ParseKxxDocumentBalanceDescription(string input)
+    {
+        const int lineNumLen = 4;
+
+        string pattern = $@"^G/\$([0-9]{{{lineNumLen}}})(.{{{_documentNumberLen}}})(.+)$";
+        Match match = Regex.Match(input, pattern);
+
+        if (!match.Success)
+        {
+            ThrowParserException($"invalid format of G/$ line. Found: {input}. Expected format: G/$rrrrccccccccctttttttttttttttttttttttttttttttttttttttt...");
+        }
+
+        if (!uint.TryParse(match.Groups[1].Value, out uint lineNum))
+        {
+            ThrowParserException($"invalid format of G/$ line. Failed to parse line number {match.Groups[1].Value}");
+        }
+        if (!uint.TryParse(match.Groups[2].Value, out uint documentNumber))
+        {
+            ThrowParserException($"invalid format of G/@ line. Failed to parse document number {match.Groups[2].Value}");
+        }
+        string description = match.Groups[3].Value;
+
+        return new KxxDocumentBalanceDescription(
+            DocumentLineNumber: lineNum,
+            DocumentId: documentNumber,
+            BalanceDescription: description);
+    }
+
+    /// <summary>
+    /// Parses G/# line in .kxx file - document description
+    /// G/$rrrrccccccccctttttttttttttttttttttttttttttttttttttttt...
+    /// </summary>
+    /// <param name="input"></param>
+    /// <returns></returns>
+    internal KxxDocumentDescription ParseKxxDocumentDescription(string input)
+    {
+        const int lineNumLen = 4;
+
+        string pattern = $@"^G/#([0-9]{{{lineNumLen}}})(.{{{_documentNumberLen}}})(.+)$";
+        Match match = Regex.Match(input, pattern);
+
+        if (!match.Success)
+        {
+            ThrowParserException($"invalid format of G/# line. Found: {input}. Expected format: G/#rrrrccccccccctttttttttttttttttttttttttttttttttttttttt...");
+        }
+
+        if (!uint.TryParse(match.Groups[1].Value, out uint lineNum))
+        {
+            ThrowParserException($"invalid format of G/# line. Failed to parse line number {match.Groups[1].Value}");
+        }
+        if (!uint.TryParse(match.Groups[2].Value, out uint documentNumber))
+        {
+            ThrowParserException($"invalid format of G/# line. Failed to parse document number {match.Groups[2].Value}");
+        }
+
+        string descriptionsString = match.Groups[3].Value;
+        if (!descriptionsString.StartsWith('*'))
+        {
+            // LogWarning($"Document description in non-dictionary format: {input}");
+            return new KxxDocumentDescription(
```

Comment:
Jak to funguje v situaci, kdy je vice radek Descriptions - napr. jedna ta plain textova a nekolik tech slovnikovych? Jak se to slozi dohromady?